src/Sql/Sql/ManagedInstance/Cmdlet/SetAzureSqlManagedInstance.cs

```diff
@@ -187,6 +206,60 @@ protected override IEnumerable<Model.AzureSqlManagedInstanceModel> GetEntity()
             return new List<AzureSqlManagedInstanceModel>() { ModelAdapter.GetManagedInstance(this.ResourceGroupName, this.Name) };
         }
 
+        /// <summary>
+        /// Validate requested Hardware family.
+        /// </summary>
+        protected bool ShouldConfirmHardwareFamilyChange()
+        {
+            bool shouldConfirmHardwareFamilyChange = false;
+
+            ModelAdapter = InitModelAdapter();
+            AzureSqlManagedInstanceModel existingInstance = ModelAdapter.GetManagedInstance(this.ResourceGroupName, this.Name);
+
+            // Get current hardware family
+            string currentHardwareFamily = existingInstance.Sku.Family;
+
+            // Get requested hardware family
+            string requestedHardwareFamily = !string.IsNullOrWhiteSpace(this.SkuName) ? AzureSqlManagedInstanceAdapter.GetHardwareGenerationFromSkuName(this.SkuName) : this.ComputeGeneration;
+
+            bool isHardwareFamilyChanged = !string.IsNullOrWhiteSpace(requestedHardwareFamily) && !currentHardwareFamily.Equals(requestedHardwareFamily, StringComparison.InvariantCultureIgnoreCase);
+
+            // Check whether hardware family is being changed to a deprecated hardware family
```

Comment:
Don't you mean, check whether it's being changed **away from** a deprecated family?

Reply:
I think this code is wrong. Instead of checking if requested == Gen5, it should be checking if current == Gen4. Moving Gen7->Gen5 shouldn't require confirmation, right?

Reply:
You are absolutely correct. Brilliant catch. Fixing this